core/vm/instructions.go

```diff
@@ -598,8 +596,11 @@ func opCreate(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]b
 	scope.Contract.Gas += returnGas
 
 	if suberr == ErrExecutionReverted {
+		// set the return data
+		interpreter.returnData = res
 		return res, nil
 	}
+	interpreter.returnData = nil
```

Comment:
In general I'm not happy how `returnData` is handled after these changes, but wanted to see big impact all changes have. Also please note that returning `[]byte` from instructions is not needed any more.

Reply:
In `CREATE` and `CREATE2` the `res` is either returned bytecode or revert return data, depending on the error. So we need handle `ErrExecutionReverted` separately. But I think it would be better to reset `returnData = nil` before `evm.Create()` and only assign it in the end in case of `ErrExecutionReverted`.

I can also move the `returnData` change to separate PR.

Reply:
I left it with some better comments.